javagen/src/main/java/com/azure/autorest/template/StreamSerializationModelTemplate.java

```diff
@@ -291,7 +291,9 @@ private static void serializeJsonProperty(JavaBlock methodBlock, ClientModelProp
                 methodBlock.line(fieldSerializationMethod + ";");
             }
         } else if (wireType == ClassType.Object) {
-            methodBlock.line("jsonWriter.writeUntyped(" + propertyValueGetter + ");");
+            methodBlock.line("jsonWriter.writeUntypedField(\"" + serializedName + "\", " + propertyValueGetter + ");");
+        } else if (wireType == ClassType.BinaryData) {
+            methodBlock.line("jsonWriter.writeUntypedField(\"" + serializedName + "\", " + propertyValueGetter + ".toObject(Object.class));");
```

Comment:
I've changed the `Object` path as well. I think previous `writeUntyped` may be a bug.